test/connect.cpp

```diff
@@ -603,7 +599,22 @@ BOOST_AUTO_TEST_CASE( noclean ) {
                 case 2:
                     MQTT_CHK("h_close3");
                     c->set_clean_session(false);
-                    c->connect();
+                    switch (c->get_protocol_version()) {
+                    case MQTT_NS::protocol_version::v3_1_1:
+                        c->connect();
+                        break;
+                    case MQTT_NS::protocol_version::v5:
+                        // set session_expiry_interval as infinity.
+                        c->connect(
+                            std::vector<MQTT_NS::v5::property_variant>{
+                                MQTT_NS::v5::property::session_expiry_interval(0xFFFFFFFFUL)
```

Comment:
Is this infinity, or just maximum? It's a uint16_t, right?

Reply:
Ah, it's actually infinity.

3.1.2.11.2 Session Expiry Interval
17 (0x11) Byte, Identifier of the Session Expiry Interval.

Followed by the Four Byte Integer representing the Session Expiry Interval in seconds. It is a Protocol Error to include the Session Expiry Interval more than once.

 

If the Session Expiry Interval is absent the value 0 is used. If it is set to 0, or is absent, the Session ends when the Network Connection is closed.

 

If the Session Expiry Interval is 0xFFFFFFFF (UINT_MAX), the Session does not expire.

 

The Client and Server MUST store the Session State after the Network Connection is closed if the Session Expiry Interval is greater than 0 [MQTT-3.1.2-23].